api/pkg/transformer/pipeline/util.go

```diff
@@ -16,6 +18,51 @@ func evalJSONPath(env *Environment, jsonPath string) (interface{}, error) {
 }
 
 func evalExpression(env *Environment, expression string) (interface{}, error) {
+	val, err := getVal(env, expression)
+	if err != nil {
+		return nil, err
+	}
+	switch exprVal := val.(type) {
+	case *operation.OperationNode:
+		val, err = exprVal.Execute()
```

Comment:
Is there any compilation penalty with this?

Reply:
what do you mean by `compilation penalty`?

Reply:
Nevermind, I thought `exprVal.Execute()` is calling expr `Execute`